dbaccess/context.go

```diff
@@ -11,24 +11,28 @@ type Context interface {
 	accessor() (database.DataAccessor, error)
 }
 
-type noTxContext struct{}
-
-var noTxContextSingleton = &noTxContext{}
-
-func (*noTxContext) accessor() (database.DataAccessor, error) {
-	return db()
+type noTxContext struct {
+	backend *DatabaseContext
 }
 
-// NoTx creates and returns an instance of dbaccess.Context without an attached database transaction
-func NoTx() Context {
-	return noTxContextSingleton
+func (ctx *noTxContext) accessor() (database.DataAccessor, error) {
```

Comment:
Do we maybe want to get rid of this function and implement it directly as part of `DatabaseContext`?

Reply:
Nope, it makes the API of noTxContext and TxContext similar, allowing the rest of dbaccess function to be agonstic to tx/noTx